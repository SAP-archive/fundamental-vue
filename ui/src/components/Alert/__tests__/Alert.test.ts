import { assert } from "chai";
import { mount, shallowMount } from "@vue/test-utils";
import { Alert } from "../";

describe("Alert", () => {
  it("renders default slot when passed", () => {
    const dummySlot = "Slot text";
    const wrapper = shallowMount(Alert, {
      slots: {
        default: dummySlot
      }
    });
    assert.include(wrapper.text(), dummySlot);
  });
  it("click on close hides the alert", () => {
    const alert = mount(Alert, {
      propsData: {
        dismissible: true
      }
    });
    assert(alert.isVisible());
    const closeButton = alert.find("button");
    assert.isDefined(closeButton);
    closeButton.trigger("click");
    assert(alert.isVisible() === false);
  });
});
