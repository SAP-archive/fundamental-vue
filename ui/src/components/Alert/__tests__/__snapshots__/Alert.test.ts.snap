// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<div
  class="fd-alert fd-alert--warning"
  id="warningAlert"
  name="fade"
  role="alert"
>
  <button
    aria-controls="alert"
    aria-label="Close"
    class="fd-alert__close"
  />
   
</div>
`;

exports[`Alert renders correctly 2`] = `
<div
  class="fd-alert fd-alert--error fd-alert--dismissible"
  id="errorAlert"
  name="fade"
  role="alert"
>
  <!---->
   
</div>
`;
